lib/vector-core/buffers/src/buffer_usage_data.rs

```diff
@@ -0,0 +1,87 @@
+use core_common::internal_event::emit;
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::{sync::Arc, time::Duration};
+use tokio::time::interval;
+use tracing::{Instrument, Span};
+
+use crate::internal_events::{BufferEventsReceived, BufferEventsSent, EventsDropped};
+
+pub struct BufferUsageData {
+    received_event_count: AtomicUsize,
+    received_event_byte_size: AtomicUsize,
+    sent_event_count: AtomicUsize,
+    sent_event_byte_size: AtomicUsize,
+    dropped_event_count: Option<AtomicUsize>,
```

Comment:
Do prefer a fixed-width integer here, compared to the `usize` that will vary based on machine word size. Ideally we'd only use `usize` in situations where we're indexing into memory or into array-like structures.

Reply:
This might be awkward if your incoming counts are `usize`, of course.

Reply:
I see, that makes sense. I'd chosen `usize` to match the `internal_event` fields I'd defined but didn't have a strong reason for `usize` in the first place. I'll make adjustments here.